apiextensions/storageversion/migrator.go

```diff
@@ -29,20 +29,23 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/tools/pager"
+	"knative.dev/pkg/logging"
 )
 
 // Migrator will read custom resource definitions and upgrade
 // the associated resources to the latest storage version
 type Migrator struct {
-	dynamicClient dynamic.Interface
-	apixClient    apixclient.Interface
+	dynamicClient  dynamic.Interface
+	apixClient     apixclient.Interface
+	ignoreNotFound bool
 }
 
 // NewMigrator will return a new Migrator
-func NewMigrator(d dynamic.Interface, a apixclient.Interface) *Migrator {
+func NewMigrator(d dynamic.Interface, a apixclient.Interface, i bool) *Migrator {
```

Comment:
Since `IgnoreNotFound` is optional I think the best approach would be a public property on the struct and you can set it _after_ calling `NewMigrator`

ie 
```go
migrator := NewMigrator(d, a)
migrator.IgnoreNotFound = true
```

This would ensure API compatibility - and we don't break existing consumers of this package